config/initializers/secure_headers.rb

```diff
@@ -68,3 +68,53 @@
   #   ]
   # }
 end
+
+# A tiny middleware that calls a block on each request. When both:
+# 1) the block returns true
+# 2) the response is a 2XX response
+# It deletes the Content-Security-Policy header. This is intended so that we can override
+# SecureHeaders behavior and not set the headers on asset files, because the headers should be set
+# on the document that links to the assets, not the assets themselves.
+class SecureHeaders::RemoveContentSecurityPolicy
+  # @yieldparam [Rack::Request] request
+  def initialize(app, &block)
+    @app = app
+    @block = block
+  end
+
+  def call(env)
+    status, headers, body = @app.call(env)
+
+    if (200...300).cover?(status) && @block.call(Rack::Request.new(env))
+      headers.delete('Content-Security-Policy')
+    end
+
+    [status, headers, body]
+  end
+end
+
+# We need this to be called after the SecureHeaders::Railtie adds its own middleware at the top
+Rails.application.configure do |config|
+  # I18n is not configured yet at this point
+  available_locales = AppConfig.env.available_locales.try(:split, ' ') || %w[en]
+
+  # example URLs:
+  # - /verify/doc_auth/AcuantImageProcessingWorker.min.js
+  # - /en/verify/capture_doc/AcuantImageProcessingWorker.min.js
+  acuant_sdk_static_files = [nil, *available_locales].product(
+                              %w[doc_auth capture_doc],
+                              %w[
+                                AcuantImageProcessingWorker.min.js
+                                AcuantImageProcessingWorker.wasm
+                              ],
```

Comment:
In doing some testing on my phone, it appears the only one of these strictly required to have the CSP header is the `.min.js` file, as otherwise there's an error about "creating a WASM object".

```suggestion
                              %w[AcuantImageProcessingWorker.min.js]
```

Reply:
ok so I'll update the specs to expect that and change the loop up

Reply:
Ok so I pushed 738414ff62ccf739de92d23fd4a8d54ce61bbcde which implements that and simplifies

Also regrouped the specs so that when run with `--format documentation` they print out clear URLs that we are testing:

```
requesting acuant SDK assets
  /verify/doc_auth/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /verify/doc_auth/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /verify/doc_auth/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /verify/doc_auth/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /verify/capture_doc/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /verify/capture_doc/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /verify/capture_doc/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /verify/capture_doc/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /en/verify/doc_auth/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /en/verify/doc_auth/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /en/verify/doc_auth/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /en/verify/doc_auth/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /en/verify/capture_doc/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /en/verify/capture_doc/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /en/verify/capture_doc/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /en/verify/capture_doc/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /es/verify/doc_auth/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /es/verify/doc_auth/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /es/verify/doc_auth/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /es/verify/doc_auth/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /es/verify/capture_doc/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /es/verify/capture_doc/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /es/verify/capture_doc/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /es/verify/capture_doc/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /fr/verify/doc_auth/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /fr/verify/doc_auth/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /fr/verify/doc_auth/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /fr/verify/doc_auth/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
  /fr/verify/capture_doc/AcuantImageProcessingWorker.min.js
    renders a JS asset
    does not include a CSP header
    does not include a session
  /fr/verify/capture_doc/AcuantImageProcessingWorker.wasm
    renders a WASM asset
    includes a CSP header with unsafe-eval
    does not include a session
  /fr/verify/capture_doc/something-that-does-not-exist/AcuantImageProcessingWorker.wasm (invalid asset)
    renders a 404 and leaves in the CSP headers
  /fr/verify/capture_doc/AcuantImageProcessingService.wasm.map (map file)
    renders a 404 and leaves in the CSP headers
```

Reply:
and I flip-flopped the asset again, fixed in 3f12bcadfbcb5c2c252214d6b13b2d49aa43a8f5